gtsfm/ui/process_graph_generator.py

```diff
@@ -0,0 +1,167 @@
+"""
+Creates a process graph based on the classes that subclass GTSFMProcess.
+
+Specifically, all classes that subclass GTSFMProcess are added to a central
+registry on declaration. GTSFMProcess has the abstract class method
+get_ui_metadata(), which returns a UiMetadata object. 
+
+ProcessGraphGenerator combines all the UiMetadata of all the GTSFMProcesses into
+one graph of blue and gray nodes, then saves to a file.
+
+Author: Kevin Fu
+"""
+
+from gtsfm.ui.registry import RegistryHolder
+
+from pathlib import Path
+import os
+
+import pydot
+from collections import namedtuple
+
+from typing import Set
+from gtsfm.ui.gtsfm_process import UiMetadata
+
+JS_ROOT = os.path.join(Path(__file__).resolve().parent.parent.parent, "rtf_vis_tool")
+DEFAULT_GRAPH_VIZ_OUTPUT_PATH = os.path.join(JS_ROOT, "src", "ui", "process_graph_output.svg")
+
+
+class ProcessGraphGenerator:
+    """Generates and saves a graph of all the components in REGISTRY."""
+
+    def __init__(self, test_mode: bool = False) -> None:
+        """Create ProcessGraphGenerator.
+
+        Args:
+            test_mode: boolean flag, only set True when unit testing.
+        """
+
+        # create empty directed graph
+        self._main_graph = pydot.Dot(graph_type="digraph", fontname="Veranda", bgcolor="white")
+
+        # style constants, see http://www.graphviz.org/documentation/
+        style_consts = {
+            "blue_fillcolor": "lightskyblue",
+            "gray_fillcolor": "gainsboro",
+            "node_style": '"rounded, filled, solid"',
+            "node_shape": "box",
+            "arrow_color": "gray75",
+        }
+        # namedtuple here is faster than default Dict and allows use of dot
+        # operator
+        StyleTuple = namedtuple("StyleTuple", style_consts)
```

Comment:
If we only use this NamedTuple definition once in this file, it could make sense to just make these global constants we use in this file

```python
BLUE_FILLCOLOR = "lightskyblue"
GRAY_FILLCOLOR = "gainsboro"
NODE_STYLE = '"rounded, filled, solid"'
NODE_SHAPE = "BOX"
ARROW_COLOR = "gray75"
```

Are we still using the "gray" and "blue" terminology though, or "PRODUCT" vs "PROCESS"?

Reply:
Good point. Updated to be globals and use "PRODUCT_FILLCOLOR"/"PROCESS_FILLCOLOR"